python_modules/libraries/dagster-dbt/dagster_dbt/cloud/resources.py

```diff
@@ -810,3 +823,90 @@ def fetch_workspace_data(self) -> DbtCloudWorkspaceData:
+    def load_specs(
+        self, dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+    ) -> Sequence[Union[AssetSpec, AssetCheckSpec]]:
+        dagster_dbt_translator = dagster_dbt_translator or DagsterDbtTranslator()
+
+        with self.process_config_and_initialize_cm() as initialized_workspace:
+            defs = DbtCloudWorkspaceDefsLoader(
+                workspace=initialized_workspace,
+                translator=dagster_dbt_translator,
+            ).build_defs()
+            asset_specs = check.is_list(
+                defs.assets,
+                AssetSpec,
+            )
+            asset_check_specs = check.is_list(
+                [
+                    check_spec
+                    for asset_def in defs.asset_checks
+                    for check_spec in asset_def.check_specs
+                ],
+                AssetCheckSpec,
+            )
+            return [*asset_specs, *asset_check_specs]
+
+    def load_asset_specs(
+        self, dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+    ) -> Sequence[AssetSpec]:
+        return [
+            spec
+            for spec in self.load_specs(dagster_dbt_translator=dagster_dbt_translator)
+            if isinstance(spec, AssetSpec)
+        ]
+
+    def load_check_specs(
+        self, dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+    ) -> Sequence[AssetCheckSpec]:
+        return [
+            spec
+            for spec in self.load_specs(dagster_dbt_translator=dagster_dbt_translator)
+            if isinstance(spec, AssetCheckSpec)
+        ]
+
+
+@preview
+def load_dbt_cloud_asset_specs(
+    workspace: DbtCloudWorkspace, dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+) -> Sequence[AssetSpec]:
+    return workspace.load_asset_specs(dagster_dbt_translator=dagster_dbt_translator)
+
+
+@preview
+def load_dbt_cloud_check_specs(
+    workspace: DbtCloudWorkspace, dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+) -> Sequence[AssetCheckSpec]:
+    return workspace.load_check_specs(dagster_dbt_translator=dagster_dbt_translator)
+
+
+@preview
+@record
+class DbtCloudWorkspaceDefsLoader(StateBackedDefinitionsLoader[DbtCloudWorkspaceData]):
+    workspace: DbtCloudWorkspace
+    translator: DagsterDbtTranslator
+
+    @property
+    def defs_key(self) -> str:
+        return f"{DBT_CLOUD_RECONSTRUCTION_METADATA_KEY_PREFIX}.{self.workspace.unique_id}"
+
+    def fetch_state(self) -> DbtCloudWorkspaceData:
+        return self.workspace.fetch_workspace_data()
+
+    def defs_from_state(self, state: DbtCloudWorkspaceData) -> Definitions:
+        all_asset_specs, all_check_specs = build_dbt_specs(
+            manifest=state.manifest,
+            translator=self.translator,
+            select="fqn:*",
+            exclude="",
+            io_manager_key=None,
+            project=None,
+        )
+
+        @multi_asset_check(specs=all_check_specs)
+        def _all_asset_checks(): ...
```

Comment:
add a comment / linear issue that it's currently impossible to define external facing checks